pkg/gencred/gencred.go

```diff
@@ -28,23 +28,120 @@ import (
 	"time"
 
 	certificates "k8s.io/api/certificates/v1beta1"
-	v1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	"k8s.io/client-go/util/cert"
 	"k8s.io/client-go/util/keyutil"
 )
 
 const (
+	// clusterRoleName is the name for the superuser ClusterRole.
+	clusterRoleName = "cluster-admin"
```

Comment:
This is only used in one location. Please do not do this. It is a giant waste of brain power. 

a) If you're asking my brain to try and remember the existence of a clusterRoleName const, please ensure it is used in at least 3 places (or maybe two). If this isn't the case, just use a string literal.
b) If you're only using the const inside a single function, please declare the const inside the function, not at package level.
c) thank you for not exporting this constant :)

Reply:
Updated. Thanks for the feedback and tips!